src/Components/WeatherDay/WeatherDay.jsx

```diff
@@ -1,240 +1,138 @@
-
-  // TODO: class names are very long. An element is always part of a block, not another element
-  // https://en.bem.info/methodology/quick-start/#nesting-1
-
+  const currentDay = useMemo(
+    () =>
+      `${dayjs(currentDayData.time).format("DD-MMMM-YYYY")} ${dayjs().format(
+        "HH"
+      )} : ${dayjs().format("mm")}`,
+    []
+  );
   return (
     <>
-      <div className="main-part-container block-color">
-        <p className="main-part-container__day text-color ff-openSans-reg">
+      <div className="wd-container block-color">
+        <p className="wd-container__day text-color ff-openSans-reg">
           {currentDay}
         </p>
-        <div className="weather-container">
-          <div className="weather-container__main">
-            <div className="weather-container__main-img">
+        <div className="wd-container__main">
+          <div className="wd-container__left-side">
+            <div className="wd-container__left-img">
               <img
                 src={`./assets/img/${currentWeatherImg}`}
                 alt="weather-img"
               />
             </div>
-            <p className="weather-container__main-text text-color ff-inter-reg">
+            <p className="wd-container__left-text text-color ff-inter-reg">
               {`${currentDayTemp < 0 ? "-" : "+"} ${currentDayTemp}`}°C
             </p>
           </div>
-          <div className="weather-container__weather-during-day">
-            <table className="weather-container__table text-color">
+          <div className="wd-container__right-side">
+            <table className="wd-container__table text-color">
               <thead>
                 <tr>
                   <th colSpan="2"> </th>
                   <th> </th>
-                  <th
-                    colSpan="3"
-                    className="weather-container__table-day-period-name"
-                  >
-                    Night
-                  </th>
-                  <th> </th>
-                  <th
-                    colSpan="3"
-                    className="weather-container__table-day-period-name"
-                  >
-                    Morning
-                  </th>
-                  <th> </th>
-                  <th
-                    colSpan="3"
-                    className="weather-container__table-day-period-name"
-                  >
-                    Day
-                  </th>
-                  <th> </th>
-                  <th
-                    colSpan="3"
-                    className="weather-container__table-day-period-name"
-                  >
-                    Evening
-                  </th>
+                  {["Night", "Morning", "Day", "Evening"].map(
+                    (dayPeriod, index) => (
+                      <WeatherDayPeriod
+                        dayPeriod={dayPeriod}
+                        index={index}
+                        key={index}
+                      />
```

Comment:
Great work!) You can also use a `dayPeriod` as a key in this case because it is also a unique static value